### PR TITLE
set command supports more flags

### PR DESCRIPTION
**Pull request by siddontang**

### CODE REVIEW

File: pkg/store/string.go [hunk 1]
```diff
@@ -155,6 +220,7 @@ func (s *Store) Set(db uint32, args [][]byte) error {
 
 	o := newStringRow(db, key)
 	o.Value = value
+	o.ExpireAt = expireat
```
Comment (qiuyesuifeng) [inline pkg/store/string.go]: if no EX or PX in set cmd,  expireat should be original o.ExpireAt value not int64(0)?

File: pkg/service/string_test.go [hunk 2]
```diff
@@ -73,10 +73,20 @@ func (s *testServiceSuite) TestXIncrByFloat(c *C) {
...
+	s.checkString(c, "hello", "get", k)
+	s.checkOK(c, "set", k, "hello1", "XX")
+	s.checkString(c, "hello1", "get", k)
+	s.checkBytes(c, nil, "SET", k, "hello", "NX")
+	s.checkInt(c, 1, "del", k)
+	s.checkBytes(c, nil, "SET", k, "hello", "XX")
+	s.checkOK(c, "set", k, "hello", "NX")
+	s.checkString(c, "hello", "get", k)
+	s.checkBytes(c, nil, "set", k, "hello", "EX", 1000, "NX")
+	s.checkOK(c, "set", k, "hello", "EX", 2000, "XX")
```
Comment (qiuyesuifeng) [inline pkg/service/string_test.go]: can EX and PX appear in one set cmd like "set a 100 ex 10 px 10000"?

Comment (siddontang) [inline pkg/service/string_test.go]: redis supports this, but will use the last one, here will use "px 10000"

File: pkg/store/string.go (hunk 1 above)
Comment (siddontang) [inline pkg/store/string.go]: set will clear expiration if no EX and PX.

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**